pkg/middleware/middleware.go

```diff
@@ -231,11 +232,26 @@ func WriteSessionCookie(ctx *m.ReqContext, value string, maxLifetimeDays int) {
 }
 
 func AddDefaultResponseHeaders() macaron.Handler {
-	return func(ctx *m.ReqContext) {
-		if ctx.IsApiRequest() && ctx.Req.Method == "GET" {
-			ctx.Resp.Header().Add("Cache-Control", "no-cache")
-			ctx.Resp.Header().Add("Pragma", "no-cache")
-			ctx.Resp.Header().Add("Expires", "-1")
+	return func(ctx *macaron.Context) {
+		if ctx.Req.Method != http.MethodGet {
+			return
+		}
+
+		if m.IsApiRequest(ctx.Req.URL.Path) {
```

Comment:
Why the distinction? Looks good otherwise

Reply:
We want to set no cache for everything requested from /api, but only for html content which is not /api. Are you thinking this can be written in a one liner or are you referring to why `ctx.Resp.Before` is not used for adding headers for api requests?

Reply:
Decided to put no cache for all non-static content